src/sentry/static/sentry/app/utils/profiler.jsx

```diff
@@ -19,11 +19,19 @@ export default function profiler() {
         this.finishProfile();
       }
 
-      activity = Integrations.Tracing.pushActivity(displayName, {
-        data: {},
-        op: 'react',
-        description: `<${displayName}>`,
-      });
+      activity = Integrations.Tracing.pushActivity(
+        displayName,
+        {
+          data: {},
+          op: 'react',
+          description: `<${displayName}>`,
+        },
+        {
+          autoPopAfter: 500, // After this timeout we'll pop this activity regardless
```

Comment:
@billyvg would this timeout be sufficient?

Reply:
@HazAT Does this mean if the react component is waiting for a request that takes longer than 500ms to finish before ending this span, it will automatically pop after 500ms?

Reply:
@billyvg Yes, exactly, so renders now never take longer than 500ms, and the span will be marked with the status `Deadline_Exceeded`.

One example when this occurred:
Sometimes when you navigation change into the project creation page in Sentry, the render activity was started but never ended, that's how we determined this "bug".
So I know this is not ideal but I had no idea how to debug why sometimes the profiler is not calling finish.